FEAT : DDM-94 AOS 로그인 유형 선택 화면 만들기

minseonglove opened the pull request:
### 개요
로그인 유형 선택 레이아웃 개발

### 변경사항
로그인 유형 선택 fragment 추가 (LoginTypeFragment.kt, fragment_login_type.xml)
로그인 화면 fragment 추가 (LoginFragment.kt, fragment_login.xml)

### 관련 지라 및 위키 링크
[DDM-94](https://fake-developers.atlassian.net/jira/software/projects/DDM/boards/10?selectedIssue=DDM-94)

### 리뷰어에게 하고 싶은 말
어서 윤허해 주십시오.

minseonglove committed:
FEAT : DDM-94 AOS 로그인 유형 선택 화면 만들기

minseonglove committed:
Create README.md

konempty commented on app/src/main/java/com/fakedevelopers/ddangddangmarket_aos/MainActivity.kt:
```diff
@@ -2,10 +2,31 @@ package com.fakedevelopers.ddangddangmarket_aos
+    fun setFragment(name : String) {
+        val transition = supportFragmentManager.beginTransaction()
+        when(name){
+            "LoginType" -> {
```
하드코딩은 좋지않아요. 최소한 상수라도 만들어 주세요. 더 좋은 선택지는 enum이 좋을듯?

konempty commented on app/src/main/java/com/fakedevelopers/ddangddangmarket_aos/ui/login/LoginFragment.kt:
```diff
@@ -0,0 +1,23 @@
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+    }
+}
```
파일끝 개행 어디감?

konempty commented on app/src/main/java/com/fakedevelopers/ddangddangmarket_aos/ui/login_type/LoginTypeFragment.kt:
```diff
@@ -0,0 +1,29 @@
+            mainActivity.setFragment("Login")
+        }
+    }
+}
```
파일끝 개행

konempty commented on app/src/main/res/layout/fragment_login_type.xml:
```diff
@@ -0,0 +1,71 @@
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/imageView_logintype_logo"
+        app:layout_constraintVertical_bias="0.05
+" />
```
뭐지 나 불편하게 만드는건가?

minseonglove replied:
> 뭐지 나 불편하게 만드는건가?

이건 심했네요 인정합니다

konempty commented on app/src/main/res/layout/fragment_login.xml:
```diff
@@ -0,0 +1,79 @@
+    android:layout_height="match_parent"
+    android:paddingStart="@dimen/fragment_horizontal_margin"
+    android:paddingEnd="@dimen/fragment_horizontal_margin"
+    tools:context=".ui.login.LoginFragment">
```
흠 내가 보기엔 이 줄은 왜 쓰는지 모르고 있을것 같은데 함 공부해보세요 ㅎ

minseonglove replied:
override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        _binding = FragmentLoginTypeBinding.inflate(inflater, container, false)
        return binding.root
    }

View 객체 따오는걸 코틀린 씬텍스에서 binding 방식으로 바꿔주면서
이게 어떻게 내가 쓰는 레이아웃을 딱딱 잡아주지 라는 것에 대한 심도 깊은 고민을 한 결과
tools:context가 잡아줬다는걸 알게됐읍니다.

konempty commented on app/src/main/res/values/strings.xml:
```diff
@@ -1,3 +1,18 @@
+    <string name="welcome">"Welcome!"</string>
+    <string name="invalid_username">Not a valid username</string>
+    <string name="invalid_password">Password must be >5 characters</string>
+    <string name="login_failed">"Login failed"</string>
```
오 여기는 영어로 쓰는 이유가 있나?? 그리고 이런건 IOS랑 맞춰서 개발하는거 맞지?

konempty replied:
+ FE까지 맞춰야됨

minseonglove replied:
> 오 여기는 영어로 쓰는 이유가 있나?? 그리고 이런건 IOS랑 맞춰서 개발하는거 맞지?

맞다 성현이랑 깊은 대화를 안나눴네

konempty commented:
https://github.com/FakeDevelopers/DDangDDangMarket_AOS/blob/143d485b5fde4b614f4cdf3177ced3c68f92687d/app/build.gradle#L11 앱이름에서 _aos는 빼야댐 누가봐도 안드로이드 앱인데 그걸 굳이 적어둘 필요는 없지

konempty commented:
기초적인 부분이라서 그런가 내가 손댈만한 코드는 없었고 코드 컨벤션같은 부분에서 밖에 할말이 읎네ㅋㅋㅋㅋ

konempty commented:
다 고치고 다시 불러주세요~

konempty commented:
나 왜 이 레포에 권한없음?? 1명이상 승인해줘야 머지 가능하게 바꾸려고 하는데

minseonglove committed:
REFACTOR : konempty PR Review 반영

소스코드 파일에 마지막 개행 추가, setFragment의 매개변수를 String이 아닌 FragmentList enum을 받도록 수정, 사용하지 않는 문자열 리소스 제거, View와 변수를 연결하는 방식을 Kotlin Synthetic에서 View Binding으로 변경

minseonglove committed:
Merge remote-tracking branch 'origin/feature/DDM-94' into feature/DDM-94

konempty commented on app/src/main/java/com/fakedevelopers/ddangddangmarket/FragmentList.kt:
```diff
@@ -0,0 +1,5 @@
+package com.fakedevelopers.ddangddangmarket
+
+enum class FragmentList {
```
Fragment리스트 아니죠ㅋㅋㅋㅋ type나 typeEnum정도가 맞겠네요

konempty commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,18 +1,17 @@
-</androidx.constraintlayout.widget.ConstraintLayout>
\ No newline at end of file
+    </FrameLayout>
+</LinearLayout>
```
머리 박자 :)

konempty commented on app/src/main/res/layout/fragment_login.xml:
```diff
@@ -0,0 +1,78 @@
+        app:layout_constraintTop_toBottomOf="@+id/input_login_password"
+        app:layout_constraintVertical_bias="0.0" />
+
+</androidx.constraintlayout.widget.ConstraintLayout>
```
이야 개행 안고쳤네 :)

minseonglove committed:
FEAT : DDM-94 AOS 로그인 유형 선택 화면 만들기

konempty PR 리뷰 반영

konempty commented on app/src/androidTest/java/com/fakedevelopers/ddangddangmarket/ExampleInstrumentedTest.kt:
```diff
@@ -21,4 +21,4 @@ class ExampleInstrumentedTest {
         val appContext = InstrumentationRegistry.getInstrumentation().targetContext
         assertEquals("com.fakedevelopers.ddangddangmarket_aos", appContext.packageName)
```
이건 안바꾸시나요?

konempty commented on app/src/main/AndroidManifest.xml:
```diff
@@ -10,7 +10,7 @@
         android:theme="@style/Theme.DDangDDangMarket_AOS">
         <activity
-            android:name=".MainActivity"
+            android:name="com.fakedevelopers.ddangddangmarket.MainActivity"
```
?? 이건 .MainActivty 인식 못하면 뭔가 잘못한거임

minseonglove replied:
> ?? 이건 .MainActivty 인식 못하면 뭔가 잘못한거임

패키지명을 변경한 직후엔 gradle부분이 수정이 안돼서 AndroidManifest가 자동으로 저렇게 잡은 것 같읍니다.
이젠 패키지명 때도 잘돌아감니다

konempty commented on app/src/main/res/values/colors.xml:
```diff
@@ -7,4 +7,4 @@
     <color name="teal_700">#FF018786</color>
     <color name="black">#FF000000</color>
     <color name="white">#FFFFFFFF</color>
```
수정 사항은 아니고 그냥 팁인데 #FFF라고만 써도 하얀색임 ㅋㅋㅋㅋ 반대로 #000은 검은색이고

minseonglove replied:
> 수정 사항은 아니고 그냥 팁인데 #FFF라고만 써도 하얀색임 ㅋㅋㅋㅋ 반대로 #000은 검은색이고

이때까지 두번씩 눌렀던 제 손구락 한테 미안해지네요

konempty reviewed (approved):
여기있는 수정사항만 고쳐서 머지하세요
